pkg/prometheus/promcfg.go

```diff
@@ -116,6 +116,36 @@ func addTLStoYaml(cfg yaml.MapSlice, namespace string, tls *v1.TLSConfig) yaml.M
 	return cfg
 }
 
+func addSafeAuthorizationToYaml(cfg yaml.MapSlice, version semver.Version, assetStoreKey string, store *assets.Store, auth *v1.SafeAuthorization) yaml.MapSlice {
+	if auth == nil || version.LT(semver.MustParse("2.26.0")) {
+		return cfg
+	}
```

Comment:
Could we notify end-user that new (post 2.26.0) auth mechanism is used, but prometheus is in incorrect (too old) version? Silently ignoring this will lead to harder debugging.

Reply:
done, what do you think?